Automattic/wp-calypso
# Desktop App: update Electron and related package versions.

Slack: p1686560269103799-slack-C7YPUHBB2
Supersedes https://github.com/Automattic/wp-calypso/pull/75516.

## Proposed Changes

This PR updates the Desktop app to use more recent versions of packages. The goal is to move off deprecated versions that have security vulnerabilities.

Key changes:
- update `electron` to 25.0.1.
- update `electron-builder` to 23.0.3.
- remove `electron-rebuild`.
- update Windows signing key.

## Testing Instructions

Ensure the following build configurations are passing in TeamCity:
  - [x] Desktop E2E tests

Additionally, check on CircleCI:
  - [x] wp-desktop-assets
  - [x] wp-desktop-mac
  - [x] wp-desktop-linux
  - [x] wp-desktop-windows

## Pre-merge Checklist
